src/contexts/Balance/Provider.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Provider } from '@ethersproject/abstract-provider';
+
+// Constants
+import { WPOKT_ADDRESS } from 'constants/index';
+
+// Context
+import { Web3Context } from 'contexts/Web3';
+import { BalanceContext } from './Context';
+
+// Utils
+import { getERC20Contract } from 'utils';
+
+export const BalanceProvider: React.FC = ({ children }) => {
+  const { address, provider } = React.useContext(Web3Context);
+  const [wpoktBalance, setWpoktBalance] = React.useState<string>();
+
+  const getBalance = async (provider: Provider, tokenAddress: string, userAddress: string): Promise<string> => {
```

Comment:
We should probably wrap this in an `useCallback` as well, or move it out to not re-declare it 😄

Reply:
Fixed in 5d379a4f41e2ee7a24388f1d0303aacf5e27d59b